ovs/lib/storagerouter.py

```diff
@@ -109,20 +109,21 @@ def get_metadata(storagerouter_guid):
                 for storagedriver_partition in disk_partition.storagedrivers:
                     claimed_space += storagedriver_partition.size if storagedriver_partition.size is not None else 0
                     _directory_used_size = 0
-                    try:
-                        output = client.run('du -Pl {0}'.format(storagedriver_partition.path))
-                    except Exception as ex:
-                        logger.error('Failed to get directory usage for {0}. {1}'.format(storagedriver_partition.path, ex))
-                    else:
+                    if client.dir_exists(storagedriver_partition.path):
                         try:
-                            _used_size, _ = output.split('\t')
-                        except ValueError as ve:
-                            logger.error('Unexpected output for directory {0}. "{1}" . {2}'.format(storagedriver_partition.path, output, ve))
+                            output = client.run('du -d 0 {0}'.format(storagedriver_partition.path))
```

Comment:
I agree with @kvanhijf. This code should be made a lot shorter. For example, sth like this:

```
if client.dir_exists(foo):
    try:
        output = client.run('du -d 0 {0}'.format(foo))
        output_size, _ = output.split('\t')
        directory_used_size = int(output_size) * 1024
    except Exception as ex:
        logger.exception('Could not find disk usage: {0}'.format(ex))
        # do something, can the code below still be used?
```

If you want separate exception messages, you could catch specific errors (CalledProcessError, ValueError, ...) and log a separate message, and you could check the length of the ```output.split(...)``` before unpacking it (which can cause a ValueError).